### PR TITLE
Fix Python 3.7/3.7 support, run tests and builds on multiple Python versions

### PR DESCRIPTION
**Pull request by reuben**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,12 +15,12 @@
 scripts = []
 
 setuptools.setup(
-    name='webdataset',
+    name="webdataset-py36",
```
Comment (JanuszL) [inline setup.py:18]: I think this is not needed.

File: .gitignore [hunk 2]
```diff
@@ -1,15 +1,12 @@
 *.pyc
 _*
 !__init__.py
```
Comment (JanuszL) [inline .gitignore:3]: Can you revert this as this seems be not related to this PR.

Comment (reuben) [reply .gitignore:3]: This is preventing adding changes to checked-in GitHub Actions in the `.github` folder.

**Comment by tmbdev** [comment]
I'm sorry this got stuck for so long. If you still want something like this, please resolve the conflicts and reopen.

**Pull request closed without merging**